### PR TITLE
Provide Method to update the URL in a RequestBuilderWrapper and maintain modifications to the BoundRequestBuilder

### PR DESCRIPTION
**Pull request by rabeyta**
See #56 for the background.

I have two commits here per my initial analysis and suggestions in #56 . Based on feedback I can keep them both, remove one, or change the PR to a more optimal solution.

edit - removed the one commit i didn't like after discussion in the issue and updated the doc for the set url method

### CODE REVIEW

**Comment by nicmunroe** [comment]
Based on our discussion in issue #56, go ahead and add the getters/setters to `RequestBuilderWrapper` (option 1) and get rid of the option 2 code. And while you're in there we might as well make everything mutable (HTTP method, circuit breaker stuff). If HTTP method turns out to really truly not be mutable then that's fine, just ignore it.

**Comment by rabeyta** [comment]
@nicmunroe Sounds good. i've updated the PR with those changes. Please take a look and let me know if you can think of any other changes needed.

File: riposte-async-http-client/src/test/java/com/nike/riposte/client/asynchttp/ning/RequestBuilderWrapperTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+        // when
+        requestBuilderWrapper.setCtx(ctx);
+
+        // then
+        assertThat(ctx).isEqualTo(requestBuilderWrapper.getCtx());
+    }
+
+    @Test
+    public void get_set_Url_works_as_expected() {
```
Comment (nicmunroe) [inline riposte-async-http-client/src/test/java/com/nike/riposte/client/asynchttp/ning/RequestBuilderWrapperTest.java]: For this and the HTTP method get/set test, we should validate that it also calls the appropriate setter on the underlying `BoundRequestBuilder`. To do that, pass in a mock `BoundRequestBuilder` when creating the `RequestBuilderWrapper` and use `verify(requestBuilder).setThing(expectedValue)`.

**Comment by nicmunroe** [comment]
Awesome. Thank you as always!

**Pull request merged**